File: ts/lcof/61.ts
function isStraight(nums: number[]): boolean {
  const n = 14
  const cnts: number[] = new Array(n).fill(0)
  for (const v of nums) {
    cnts[v] += 1
  }
  let min = null, max = 0
  for (let i = 1; i < n; i++) {
    if (cnts[i] > 1) return false
    if (cnts[i] > 0) {
      if (!min) min = i
      max = i
    }
  }
  return max - min < 5
};

File: py3/lcof/61.py
from typing import List

class Solution:
    def isStraight(self, nums: List[int]) -> bool:
        n = 14
        cnts = [0] * n
        for v in nums:
            cnts[v] += 1

        st, ed = 0, 0
        for i in range(1, n):
            if cnts[i] > 1: return False
            if cnts[i] > 0:
                if not st: st = i
                ed = i

        return ed - st < 5